applications/vessel-history/src/features/vessels/activity/vessels-activity.selectors.ts

```diff
@@ -0,0 +1,50 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { RenderedEvent, selectEventsWithRenderingInfo } from 'features/vessels/activity/vessels-activity.slice';
+import { Filters, initialState, selectFilters } from './../../profile/filters/filters.slice';
+
+export const selectFilteredEvents = createSelector(
+  [selectEventsWithRenderingInfo, selectFilters],
+  (events, filters) => {
+    const startTimestamp = new Date(filters.start).getTime()
+    const endTimestamp = new Date(filters.end).getTime()
```

Comment:
When defining a date range to filter on, remember to add the proper time on the start and end. In this case:

- `startTimestamp` should be calculated for the start date selected by the user and add the `00:00:00.000` as the time part
- `endTimestamp` should be calculated for the end date selected by the user and add the `23:59:59.999` as the time part
So that any event that took place in the startDate or endDate selected by the user is shown